Unsuccessful session user deletion fails the tests

After introducing isolated user per session in ui tests (PR #3455), ui hosts tests started to fail with following error:
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.10, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 13 items

tests/foreman/ui/test_host.py s..s.ssssss.sE

==================================== ERRORS ====================================
_______ ERROR at teardown of HostTestCase.test_positive_update_with_user _______

cls = <class 'tests.foreman.ui.test_host.HostTestCase'>

    @classmethod
    def tearDownClass(cls):
        super(UITestCase, cls).tearDownClass()
        if cls.session_user is not None:
>           cls.session_user.delete(synchronous=False)

robottelo/test.py:154: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../nailgun/nailgun/entity_mixins.py:566: in delete
    response.raise_for_status()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [422]>

    def raise_for_status(self):
        """Raises stored :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ''
    
        if 400 <= self.status_code < 500:
            http_error_msg = '%s Client Error: %s for url: %s' % (self.status_code, self.reason, self.url)
    
        elif 500 <= self.status_code < 600:
            http_error_msg = '%s Server Error: %s for url: %s' % (self.status_code, self.reason, self.url)
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           HTTPError: 422 Client Error: Unprocessable Entity for url: https://example.com/api/v2/users/779

../../env/lib/python2.7/site-packages/requests/models.py:844: HTTPError
--------------------------- Captured stdout teardown ---------------------------
2016-05-23 19:58:38 - nailgun.client - DEBUG - Making HTTP DELETE request to https://example.com/api/v2/users/779 with options {'verify': False, 'auth': ('admin', 'changeme'), 'headers': {'content-type': 'application/json'}} and no data.
2016-05-23 19:58:42 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":779,"errors":{"base":["Foreman User  is used by wamwstewojvfrlyqbfdsegneehf.ydpiovqnj3","Foreman User  is used by octpbklpci.l4jp9rerdy"]},"full_messages":["Foreman User  is used by wamwstewojvfrlyqbfdsegneehf.ydpiovqnj3","Foreman User  is used by octpbklpci.l4jp9rerdy"]}
}

=============== 4 passed, 9 skipped, 1 error in 1518.19 seconds ================
```

`Foreman User  is used by octpbklpci.l4jp9rerdy` means user is in-use by some host task. Need to either find a way to wait until task finishes or just ignore failure of user deletion, as it confuses and makes thinking that there's a problem in test, not in robottelo.
imo we just need to cover user teardown with `try .. except` block, as basically user deletion is just a bonus and we definitely should not fail tests which are  not related to user.

--- a/robottelo/test.py
+++ b/robottelo/test.py
@@ -151,7 +151,10 @@ class UITestCase(TestCase):
     def tearDownClass(cls):
         super(UITestCase, cls).tearDownClass()
         if cls.session_user is not None:
-            cls.session_user.delete(synchronous=False)
+            try:
+                cls.session_user.delete(synchronous=False)
+            except Exception as e:
+                cls.logger.warn("Unable to delete session_user %s", str(e))
 
     def setUp(self):  # noqa
         """We do want a new browser instance for every test."""

